pypsa/consistency.py

```diff
@@ -25,8 +25,17 @@ def _bus_columns(df: pd.DataFrame) -> pd.Index:
     return df.columns[df.columns.str.startswith("bus")]
 
 
+def _log_or_raise(message: str, strict: bool) -> None:
+    if strict:
+        raise ValueError(message)
+    else:
+        logger.warning(message)
```

Comment:
Can we do something like?
```suggestion
def _log_or_raise(strict, message, *args):
    if strict:
        raise ValueError(message % args)
    else:
        logger.warning(message, *args)
```
This way we keep the lazy evaluation of log messages. I guess it doesn't matter too much so far, since most variables are being evaluated and we're at the warning level. But I think it's still better to stick to it

Reply:
Also, I don't think this should be a ValueError